test/e2e/metrics_test.go

```diff
@@ -0,0 +1,115 @@
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/gavv/httpexpect.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo/test/e2e/fixtures"
+)
+
+const metricsUrl = "http://localhost:9090"
+
+// ensure basic HTTP functionality works,
+// testing behaviour really is a non-goal
+type MetricsSuite struct {
```

Comment:
I'm really impressed at how quickly you've got the hang of this test suite stuff.